### ISSUE TITLE
Golden_Method: Scalar function

### ISSUE THREAD

**Issue by cc7768** [spencerlyon2/CompEcon.jl#12]
The current version of `golden_method` doesn't work with functions that use scalars.

```
f(x::Real) = x^2

julia> golden_method(f, -1., 1.)
ERROR: MethodError: `*` has no method matching *(::Array{Float64,1}, ::Array{Float64,1})
Closest candidates are:
  *(::Any, ::Any, ::Any, ::Any...)
  *{T<:Union{Complex{Float32},Float64,Float32,Complex{Float64}},S}(::Union{DenseArray{T<:Union{Complex{Float32},Float64,Float32,Complex{Float64}},2},SubArray{T<:Union{Complex{Float32},Float64,Float32,Complex{Float64}},2,A<:DenseArray{T,N},I<:Tuple{Vararg{Union{Colon,Int64,Range{Int64}}}},LD}}, ::Union{SubArray{S,1,A<:DenseArray{T,N},I<:Tuple{Vararg{Union{Colon,Int64,Range{Int64}}}},LD},DenseArray{S,1}})
  *{TA,TB}(::Base.LinAlg.AbstractTriangular{TA,S<:AbstractArray{T,2}}, ::Union{SubArray{TB,1,A<:DenseArray{T,N},I<:Tuple{Vararg{Union{Colon,Int64,Range{Int64}}}},LD},DenseArray{TB,2},SubArray{TB,2,A<:DenseArray{T,N},I<:Tuple{Vararg{Union{Colon,Int64,Range{Int64}}}},LD},DenseArray{TB,1}})
  ...
 in power_by_squaring at intfuncs.jl:78
 in f at none:1
 in golden_method at /home/chase/.julia/v0.4/CompEcon.jl/src/optimization.jl:10
 in golden_method at /home/chase/.julia/v0.4/CompEcon.jl/src/optimization.jl:37
```

**Comment by sglyon** [comment]
Implemented in cc946d01e4cd133a2794788284efd8eb5bed07cd but we should revisit this

**Comment by sglyon** [comment]
duplicate of #7 closing
